Get rustc commit hash from channel manifest

SimonSapin opened the pull request:
… added in https://github.com/rust-lang/rust/pull/44218, instead of using the GitHub API.

Also upgrade to rustc 1.22.0-nightly (d93036a04 2017-09-07).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/18420)
<!-- Reviewable:end -->

SimonSapin committed:
Get rustc commit hash from channel manifest

… added in https://github.com/rust-lang/rust/pull/44218,
instead of using the GitHub API.

nox commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 3c0680a has been approved by `nox`

<!-- @bors-servo r=nox 3c0680a574a156441029e1f36f4d7502dd1519c0 -->

bors-servo commented:
:hourglass: Testing commit 3c0680a574a156441029e1f36f4d7502dd1519c0 with merge da2d0a5ad74eb34dcdba6dba9698f9bd50aac7ad...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9151)

SimonSapin commented:
On buildbot:

```
error: failed to run custom build command for `fontsan v0.3.2 (https://github.com/servo/fontsan#ab68da17)`
could not execute process `/Users/servo/buildbot/slave/mac-dev-unit/build/target/debug/build/fontsan-c0e0814b33162e4e/build-script-build` (never executed)
```

On Travis:

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64"   […]
  = note: collect2: error: ld returned 1 exit status
          
error: aborting due to previous error
error: Could not compile `servo`.
```

… neither with any more output that seems relevant to the error. I can’t reproduce locally, on either Linux or OS X. So… I’ve no idea. Let’s see if it’s reliable?

@bors-servo retry

bors-servo commented:
:hourglass: Testing commit 3c0680a574a156441029e1f36f4d7502dd1519c0 with merge 9aca316237e5096db224f6696dcaa258407c6d39...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/5953)

SimonSapin commented:
At least some of the failures are:

```
  │ ERROR:servo: assertion failed: distance_to_stack_bottom <= max_allowable_distance
  │ assertion failed: distance_to_stack_bottom <= max_allowable_distance (thread <unnamed>, at /home/servo/buildbot/slave/linux-rel-css/build/components/style/context.rs:658)
```

CC @bholley @julian-seward1, should we increase the stack size?

bholley commented:
I don't think stack size is the issue - we just bumped the limit, and those assertion stacks are very short in any case.

We need to understand what the various values are to understand why it isn't working.

bholley commented:
(Note that we could increase the slop from 10k, but if that fixed it, that would also be surprising enough to warrant figuring out why).

bors-servo commented:
:umbrella: The latest upstream changes (presumably #18413) made this pull request unmergeable. Please resolve the merge conflicts.

SimonSapin commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 64eb2e0ececa2651c6648caa5e86ead1dcc588d3 with merge 69bac87cfcbce22d550c995f7799ed7b1bb55cb3...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/5976)

jdm commented:
More of the same assertions.

SimonSapin commented:
Bisection in #18445 shows that https://github.com/rust-lang/rust/pull/43931 is responsible (though possibly indirectly). Maybe more inlining is making some stack frames larger? @bholley, @emilio, would that make sense?

emilio commented:
I think so, yeah.

We can confirm it marking `StyleBloom::new` and `StyleSharingCandidateCache::new` with `#[inline(never)]`.

SimonSapin committed:
Upgrade to rustc 1.22.0-nightly (d93036a04 2017-09-07)

SimonSapin committed:
Un-inline some methods initializing thread_local! …

… in order to limit stack frame sizes after extra inlining from
https://github.com/rust-lang/rust/pull/43931

See https://github.com/servo/servo/pull/18420#issuecomment-328769322

SimonSapin commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit a71fe3cbce3af272b8b83d45c23c2697d3068226 with merge 415e802a5bdb0876ec7e47c67cc0786fd1d24cfd...

SimonSapin committed:
'./mach rustup': Use alternate S3 URL that doesn’t require SNI

… to hopefully fix https://travis-ci.org/servo/servo-with-rust-nightly/ builds.

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/8843), [arm32](http://build.servo.org/builders/arm32/builds/9077), [arm64](http://build.servo.org/builders/arm64/builds/9108), [linux-dev](http://build.servo.org/builders/linux-dev/builds/9167), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/5961), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/6013), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9210), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/1665), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/1671), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/5905), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/5962), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/1691), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/1687), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/4464)
State: approved= try=True

emilio commented on components/style/bloom.rs:
```diff
@@ -121,6 +121,7 @@ impl<E: TElement> StyleBloom<E> {
     /// Create an empty `StyleBloom`. Because StyleBloom acquires the thread-
     /// local filter buffer, creating multiple live StyleBloom instances at
     /// the same time on the same thread will panic.
+    #[inline(never)]
```
Maybe leave a comment on the reasoning for this here and below?

SimonSapin replied:
Done.

SimonSapin commented:
@bors-servo r=nox,emilio

bors-servo commented:
:pushpin: Commit 052a8b5 has been approved by `nox,emilio`

<!-- @bors-servo r=nox,emilio 052a8b500af3afb0259bae2b04557facdf062a2e -->

bors-servo commented:
:hourglass: Testing commit 052a8b500af3afb0259bae2b04557facdf062a2e with merge af077a722225193b38d80622fb939b7719b46db0...
